src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts

```diff
@@ -0,0 +1,480 @@
+                            }
+                        })
+                        .always(() => this.spinners.generateCertificate(false));
+                }
+            });
+    }
+    
+    exportHubConfiguration() {
+        if (!this.isValid(this.editedHubTask().validationGroupForExport)) {
+            return;
+        }
+        
+        this.exportConfiguration();
+    }
+
+    exportAccessConfiguration() {
+        if (!this.isValid(this.editedHubTask().validationGroupForExport)) {
+            return;
+        }
+        
+        if (this.editedHubTask().withFiltering()) {
+            if (!this.isValid(this.editedReplicationAccessItem().validationGroupForExportWithFiltering)) {
+                return;
+            }
+        } else {
+            if (!this.isValid(this.editedReplicationAccessItem().validationGroupForExportNoFiltering)) {
+                return;
+            }
+        }
+        
+        this.exportConfiguration(true);
+        this.editedReplicationAccessItem().accessConfigurationWasExported(true);
+    }
+    
+    exportConfiguration(includeAccessInfo: boolean = false) {
+        const hubTaskItem = this.editedHubTask();
+        const databaseName = this.activeDatabase().name;
+        const topologyUrls = clusterTopologyManager.default.topology().nodes().map(x => x.serverUrl());
+
+        let configurationToExport = {
+            Database: databaseName,
+            HubTaskName: hubTaskItem.taskName(),
+            TopologyUrls: topologyUrls,
+            AllowHubToSinkMode: hubTaskItem.allowReplicationFromHubToSink(),
+            AllowSinkToHubMode: hubTaskItem.allowReplicationFromSinkToHub()
+        } as pullReplicationExportFileFormat;
+        
+        if (includeAccessInfo) {
+            const replicationAccessItem = this.editedReplicationAccessItem();
+            configurationToExport.AccessName = replicationAccessItem.replicationAccessName();
+            
+            // if certificate was Generated: export both public & private key
+            // if certificate was Imported:  export 'null'
+            const certificate = replicationAccessItem.certificate().certificate();
+            configurationToExport.Certificate = certificate || null;
+
+            configurationToExport.HubToSinkPrefixes = replicationAccessItem.hubToSinkPrefixes().map(x => x.path());
+            
+            if (this.editedReplicationAccessItem().samePrefixesForBothDirections()) {
+                configurationToExport.UseSamePrefixes = true;
+            } else {
+                configurationToExport.UseSamePrefixes = false;
+                configurationToExport.SinkToHubPrefixes = replicationAccessItem.sinkToHubPrefixes().map(x => x.path());
+            }
+        }
+
+        let fileName = includeAccessInfo ? "hubAccessConfiguration" : "hubConfiguration";
+        let accessName = includeAccessInfo ? this.editedReplicationAccessItem().replicationAccessName() : "";
```

Comment:
const

Reply:
if we don't include access info then export file with have `--` in name.

Reply:
done